### ISSUE TITLE
Upstream TLS 1.3 Pre-Shared Key extension parsing and writing on server side

### ISSUE THREAD

**Issue by ronald-cron-arm** [Mbed-TLS/mbedtls#5564]
This issue tracks the up-streaming of the server Pre-Shared Key extension parsing and writing.

Source: https://github.com/hannestschofenig/mbedtls/tree/tls13-prototype

multi-PSKs and 0-RTT not in scope of this issue. 

PLEASE READ THOROUGHLY section 4.2.11 of RFC 8446

Precise acceptance criteria:
Test (unit test) with an Mbed TLS client that the Mbed TLS TLS 1.3 client and server write and parse successfully Pre-Shared Key extensions (depends on #5562).

Test (ssl-opt.sh test) with an OpenSSL/GnuTLS client that:
1. the Mbed TLS TLS 1.3 server parses successfully a ClientHello containing a Pre-Shared Key extension.
2. the Mbed TLS TLS 1.3 server sends a Pre-Shared Key extension in its ServerHello that is parsed successfully by the clients.  

See test "TLS 1.3, default suite, PSK" in the prototype to do the testing.
The success of the handshake using the pre-shared key is not part of this issue.
